scripts/agent_bridge_protocol.py

```
@@ -0,0 +1,84 @@
+def encode_response_frame(payload):
+    if len(payload) > MAX_RESPONSE_PAYLOAD:
+        raise ProtocolError("response payload is too large")
+    header = MAGIC + bytes((VERSION, RESPONSE)) + len(payload).to_bytes(2, "big")
+    return header + payload + checksum(header[2:] + payload).to_bytes(2, "big")
+
+
+def decode_agent_request(payload):
+    try:
+        request = json.loads(payload.decode("utf-8"))
+    except (UnicodeDecodeError, json.JSONDecodeError) as error:
+        raise ProtocolError("request must be valid UTF-8 JSON", "invalid_request") from error
```

Ruff (0.16.1)

[warning] Avoid specifying long messages outside the exception class

(TRY003)